cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -297,18 +298,33 @@ func (c *client) CreateMachineSetObjects(machineSets []*clusterv1.MachineSet, na
 }
 
 func (c *client) CreateMachineObjects(machines []*clusterv1.Machine, namespace string) error {
+	var (
+		wg      sync.WaitGroup
+		errOnce sync.Once
+		gerr    error
+	)
+	// the approach to concurrency here comes from golang.org/x/sync/errgroup
 	for _, machine := range machines {
-		// TODO: Run in parallel https://github.com/kubernetes-sigs/cluster-api/issues/258
-		createdMachine, err := c.clientSet.ClusterV1alpha1().Machines(namespace).Create(machine)
-		if err != nil {
-			return fmt.Errorf("error creating a machine object in namespace %v: %v", namespace, err)
-		}
-		err = waitForMachineReady(c.clientSet, createdMachine)
-		if err != nil {
-			return err
-		}
+		wg.Add(1)
+
+		go func(machine *clusterv1.Machine) {
+			defer wg.Done()
+
+			createdMachine, err := c.clientSet.ClusterV1alpha1().Machines(namespace).Create(machine)
+			if err != nil {
+				errOnce.Do(func() {
+					gerr = fmt.Errorf("error creating a machine object in namespace %v: %v", namespace, err)
+				})
+				return
+			}
+			err = waitForMachineReady(c.clientSet, createdMachine)
+			if err != nil {
+				errOnce.Do(func() { gerr = err })
+			}
+		}(machine)
 	}
-	return nil
+	wg.Wait()
```

Comment:
Don't ignore the return value of Wait.

https://github.com/golang/go/wiki/CodeReviewComments#handle-errors

Reply:
Good catch. Must have missed the linter somehow.

Reply:
Nevermind, `WaitGroup.Wait()` has no return value.

Reply:
Weird. I thought it did looking at the docs but I must have been looking at the docs for the errgroup package's wait call.